client/rest-high-level/src/main/java/org/elasticsearch/client/license/GetBasicStatusResponse.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.license;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * Response class for license get basic status API
+ */
+public class GetBasicStatusResponse implements ToXContentObject {
+
+    private static final ParseField ELIGIBLE_TO_START_BASIC = new ParseField("eligible_to_start_basic");
+
+    private static final ConstructingObjectParser<GetBasicStatusResponse, Void> PARSER = new ConstructingObjectParser<>(
+        "get_basic_status_response", true, a -> new GetBasicStatusResponse((boolean) a[0]));
+
+    static {
+        PARSER.declareField(constructorArg(), (parser, context) -> parser.booleanValue(), ELIGIBLE_TO_START_BASIC,
+            ObjectParser.ValueType.BOOLEAN);
+    }
+
+    private final boolean eligibleToStartBasic;
+
+    GetBasicStatusResponse(boolean eligibleToStartBasic) {
+        this.eligibleToStartBasic = eligibleToStartBasic;
+    }
+
+    /**
+     * Returns whether the license is eligible to start basic or not
+     */
+    public boolean isEligibleToStartBasic() {
+        return eligibleToStartBasic;
+    }
+
+    public static GetBasicStatusResponse fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
this is only here for testing, and it's duplicated in the original transport client class. The problem is that `AbstractXContentTestCase` requires a type that extends `ToXContent`, and we need to test the parsing code from the client response. Do you have something in mind to be able to test the parsing code using the rendering code from Elasticsearch?

Reply:
Here is the first incarnation of what I wanted to do, and I feel like we can make this more generic but I wanted to just get it done in an idea before going crazy on a generic impl for these tests.

https://github.com/hub-cap/elasticsearch/blob/f06095cb29c72ade77dd51d62d75096071cd7619/x-pack/protocol/src/test/java/org/elasticsearch/protocol/xpack/watcher/DeleteWatchResponseTests.java

Reply:
that looks good, I am tweaking `AbstractXContentTestCase` to be able to support this scenario (especially to make sure that we don't lose testing support for unknown fields etc), but my problem is that I don't have the `org.elasticsearch.license.GetTrialStatusResponse` in the test classpath. Should I move it somewhere else? Or add the plugin to the test classpath? how did you do that in your branch?